src/porepy/numerics/fv/biot.py

```diff
@@ -1078,14 +1078,22 @@ def assemble_rhs(self, g, data):
         # successive timesteps (n and n+1) appear on the rhs with opposite signs. For
         # transient BCs, use the below with the appropriate version of d_bound_i.
         d_bound_1 = parameter_dictionary["bc_values"]
-        d_bound_0 = parameter_dictionary["bc_values"]
+
+        if "bc_values" in parameter_dictionary["state"]:
+            d_bound_0 = parameter_dictionary["state"]["bc_values"]
+        else:
+            d_bound_0 = parameter_dictionary["bc_values"]
         biot_alpha = parameter_dictionary["biot_alpha"]
         rhs_bound = (
             -matrix_dictionary["bound_div_d"] * (d_bound_1 - d_bound_0) * biot_alpha
         )
 
         # Time part
-        d_cell = parameter_dictionary["state"]
+        if "bc_values" in parameter_dictionary["state"]:
+            d_cell = parameter_dictionary["state"]["displacement"]
+        else:
+            d_cell = parameter_dictionary["state"]
```

Comment:
@IvarStefansson @keileg : 
1: What do you think about this way of setting the information from the previous time step (dictionary["state"]["'displacement"] instead of dictionary["state"]. The reason for this is that I also need access to dictionary["state"]["bc_values"] in time dependent problems)

I don't think this is the perfect way to do this, but it fits into the framework.
2:
Should we remove the if-else statement and require the parameter_dictionary["state"] to be a dictionary?

Reply:
1) I could think of alternative placements. The current is
data["PARAMETERS"]["state"]
I would say that 
data["state"]
is more natural, in a way. Or even specify to "variable_state" or something. In this case, we might also consider using upper case for consistency.
2) I agree with Eirik.

Reply:
> 1.: Do we have a structure for storing primary (and secondary) variables during time stepping? If not, I think this is a fair approach, in line with other things we are doing.
no we do not

> 2. If we go with the state, I definitely vote for requiring that the structure be followed.
I agree

> 1. I could think of alternative placements. The current is
>    data["PARAMETERS"]["state"]
>    I would say that
>    data["state"]
>    is more natural, in a way. Or even specify to "variable_state" or something. In this case, we might also consider using upper case for consistency.

Not variable_state. 1. bc_values is not a variable. 2. Not only variables are time dependent. E.g., apertures for slip, permeabilities for two phase flow...